netbsd build currently fails

1: meson.build is missing `os == netbsd` is some places (maybe use os.endswith('bsd') instead?)
2: `#include <sys/user.h>` does not exist in bsd
3: removing the header include causes other build failures wrt to undefined symbols

NetBSD NetBSD 10.1 NetBSD 10.1 (GENERIC) #0: Mon Dec 16 13:08:11 UTC 2024  mkrepro@mkrepro.NetBSD.org:/usr/src/sys/arch/amd64/compile/GENERIC amd64

```
ninja: Entering directory `build'
[1/93] Compiling C object src/shell_var/shell_var.p/shell_var.c.o
[2/93] Compiling C object src/rc-abort/rc-abort.p/rc-abort.c.o
[3/93] Linking target src/einfo/einfon
[4/93] Linking target src/einfo/einfo
[5/93] Linking target src/einfo/ewarnn
[6/93] Generating src/shared/version.h with a custom command
[7/89] Linking target src/einfo/ewarn
[8/89] Generating src/shared/version with a custom command
[9/89] Compiling C object src/mountinfo/mountinfo.p/mountinfo.c.o
FAILED: src/mountinfo/mountinfo.p/mountinfo.c.o 
cc -Isrc/mountinfo/mountinfo.p -Isrc/mountinfo -I../src/mountinfo -Isrc/shared -I../src/shared -Isrc/librc -I../src/librc -Isrc/libeinfo -I../src/libeinfo -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c99 -O0 -g -DHAVE_PAM -DRC_DEBUG -Wcast-align -Wcast-qual -Wdeclaration-after-statement -Wformat=2 -Winline -Wmissing-declarations -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wshadow -Wwrite-strings -Werror=implicit-int -Werror=implicit-function-declaration -Werror=int-conversion -DHAVE_STRLCPY -MD -MQ src/mountinfo/mountinfo.p/mountinfo.c.o -MF src/mountinfo/mountinfo.p/mountinfo.c.o.d -o src/mountinfo/mountinfo.p/mountinfo.c.o -c ../src/mountinfo/mountinfo.c
../src/mountinfo/mountinfo.c:205:4: error: 'MNT_ASYNC' undeclared here (not in a function); did you mean 'O_ASYNC'?
  205 |  { MNT_ASYNC,        "asynchronous" },
      |    ^~~~~~~~~
      |    O_ASYNC
../src/mountinfo/mountinfo.c:206:4: error: 'MNT_EXPORTED' undeclared here (not in a function)
  206 |  { MNT_EXPORTED,     "NFS exported" },
      |    ^~~~~~~~~~~~
../src/mountinfo/mountinfo.c:207:4: error: 'MNT_LOCAL' undeclared here (not in a function)
  207 |  { MNT_LOCAL,        "local" },
      |    ^~~~~~~~~
../src/mountinfo/mountinfo.c:208:4: error: 'MNT_NOATIME' undeclared here (not in a function)
  208 |  { MNT_NOATIME,      "noatime" },
      |    ^~~~~~~~~~~
../src/mountinfo/mountinfo.c:209:4: error: 'MNT_NOEXEC' undeclared here (not in a function); did you mean 'ENOEXEC'?
  209 |  { MNT_NOEXEC,       "noexec" },
      |    ^~~~~~~~~~
      |    ENOEXEC
../src/mountinfo/mountinfo.c:210:4: error: 'MNT_NOSUID' undeclared here (not in a function)
  210 |  { MNT_NOSUID,       "nosuid" },
      |    ^~~~~~~~~~
../src/mountinfo/mountinfo.c:214:4: error: 'MNT_QUOTA' undeclared here (not in a function)
  214 |  { MNT_QUOTA,        "with quotas" },
      |    ^~~~~~~~~
../src/mountinfo/mountinfo.c:215:4: error: 'MNT_RDONLY' undeclared here (not in a function); did you mean 'O_RDONLY'?
  215 |  { MNT_RDONLY,       "read-only" },
      |    ^~~~~~~~~~
      |    O_RDONLY
../src/mountinfo/mountinfo.c:216:4: error: 'MNT_SYNCHRONOUS' undeclared here (not in a function)
  216 |  { MNT_SYNCHRONOUS,  "synchronous" },
      |    ^~~~~~~~~~~~~~~
../src/mountinfo/mountinfo.c:217:4: error: 'MNT_UNION' undeclared here (not in a function)
  217 |  { MNT_UNION,        "union" },
      |    ^~~~~~~~~
../src/mountinfo/mountinfo.c:227:4: error: 'MNT_SOFTDEP' undeclared here (not in a function)
  227 |  { MNT_SOFTDEP,      "soft-updates" },
      |    ^~~~~~~~~~~
../src/mountinfo/mountinfo.c: In function 'find_mounts':
../src/mountinfo/mountinfo.c:253:15: error: implicit declaration of function 'getmntinfo' [-Werror=implicit-function-declaration]
  253 |  if ((nmnts = getmntinfo(&mnts, MNT_NOWAIT)) == 0)
      |               ^~~~~~~~~~
../src/mountinfo/mountinfo.c:253:15: warning: nested extern declaration of 'getmntinfo' [-Wnested-externs]
../src/mountinfo/mountinfo.c:253:33: error: 'MNT_NOWAIT' undeclared (first use in this function); did you mean 'WNOWAIT'?
  253 |  if ((nmnts = getmntinfo(&mnts, MNT_NOWAIT)) == 0)
      |                                 ^~~~~~~~~~
      |                                 WNOWAIT
../src/mountinfo/mountinfo.c:253:33: note: each undeclared identifier is reported only once for each function it appears in
../src/mountinfo/mountinfo.c:259:15: error: invalid use of undefined type 'struct statfs'
  259 |   flags = mnts[i].F_FLAGS & MNT_VISFLAGMASK;
      |               ^
../src/mountinfo/mountinfo.c:259:18: error: invalid use of undefined type 'struct statfs'
  259 |   flags = mnts[i].F_FLAGS & MNT_VISFLAGMASK;
      |                  ^
../src/mountinfo/mountinfo.c:259:29: error: 'MNT_VISFLAGMASK' undeclared (first use in this function)
  259 |   flags = mnts[i].F_FLAGS & MNT_VISFLAGMASK;
      |                             ^~~~~~~~~~~~~~~
../src/mountinfo/mountinfo.c:276:11: error: invalid use of undefined type 'struct statfs'
  276 |       mnts[i].f_mntfromname,
      |           ^
../src/mountinfo/mountinfo.c:276:14: error: invalid use of undefined type 'struct statfs'
  276 |       mnts[i].f_mntfromname,
      |              ^
../src/mountinfo/mountinfo.c:277:11: error: invalid use of undefined type 'struct statfs'
  277 |       mnts[i].f_mntonname,
      |           ^
../src/mountinfo/mountinfo.c:277:14: error: invalid use of undefined type 'struct statfs'
  277 |       mnts[i].f_mntonname,
      |              ^
../src/mountinfo/mountinfo.c:278:11: error: invalid use of undefined type 'struct statfs'
  278 |       mnts[i].f_fstypename,
      |           ^
../src/mountinfo/mountinfo.c:278:14: error: invalid use of undefined type 'struct statfs'
  278 |       mnts[i].f_fstypename,
      |              ^
cc1: some warnings being treated as errors
[10/89] Compiling C object src/openrc-user/openrc-user.p/openrc-user.c.o
[11/89] Compiling C object src/openrc-run/openrc-run.p/openrc-run.c.o
ninja: build stopped: subcommand failed.

```

--- a/src/librc/librc.h
+++ b/src/librc/librc.h
@@ -44,8 +44,10 @@
 #include <unistd.h>
 
 #if defined(BSD) && !defined(__GNU__)
+  #if !defined(__NetBSD__)
+  #include <sys/user.h>
+  #endif
 #include <sys/param.h>
-#include <sys/user.h>
 #include <sys/sysctl.h>
 #include <kvm.h>
 #else

--- a/src/mountinfo/mountinfo.c
+++ b/src/mountinfo/mountinfo.c
@@ -20,6 +20,11 @@
 #  include <sys/mount.h>
 
 #  define F_FLAGS f_flags
+#elif defined(__NetBSD__)
+#  include <sys/mount.h>
+
+#  define statfs statvfs
+#  define F_FLAGS f_flag
 #elif defined(BSD) && !defined(__GNU__)
 #  include <sys/statvfs.h>
 

